configure.ac

```diff
@@ -1149,6 +1149,16 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <fcntl.h>]],
  [ AC_MSG_RESULT(yes); HAVE_O_CLOEXEC=1 ],
  [ AC_MSG_RESULT(no); HAVE_O_CLOEXEC=0 ]
 )
+AC_DEFINE_UNQUOTED([HAVE_O_CLOEXEC], [$HAVE_O_CLOEXEC], [Define to 1 if O_CLOEXEC flag is available.])
+
+AC_MSG_CHECKING(for pipe2)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]],
+ [[ static int pipefd[2];
+    pipe2(pipefd, 0); ]])],
+ [ AC_MSG_RESULT(yes); HAVE_PIPE2=1 ],
+ [ AC_MSG_RESULT(no); HAVE_PIPE2=0 ]
+)
+AC_DEFINE_UNQUOTED([HAVE_PIPE2], [$HAVE_PIPE2], [Define to 1 if pipe2 is available.])
```

Comment:
```suggestion
AC_CHECK_DECLS([pipe2])
```

Reply:
Heh, nice that sometimes the solution can be so simple, thanks. Any reason why the same approach is not used for also e.g. `fdatasync`? (That's what I used as template in the previous version).